tracer/tracer.go

```diff
@@ -39,6 +57,12 @@ func NewTracerTransport(transport Transport) *Tracer {
 		buffer:              newSpansBuffer(spanBufferDefaultMaxSize),
 		sampler:             newAllSampler(),
 		DebugLoggingEnabled: false,
+
+		services:    make(map[string]*Service),
+		serviceChan: make(chan *Service),
```

Comment:
If this is not buffered, the clients might be blocked on `SetServiceInfo` during the flush.